src/test/java/junmin/netflix/costreportservice/controller/CostReportControllerTest.java

```diff
@@ -0,0 +1,211 @@
+	public void getProdCostReportByProdNameAndEP() throws Exception {
+		String epCode = "101";
+		String production = "MaryAndMartin";
+		when(mockService.getCostReportByProdNameAndEpCode(production, epCode)).thenReturn(getEPCostRecord());
+		RequestBuilder request = MockMvcRequestBuilders
+				.get("/api/report/production/"+production)
+				.param("epCode", epCode)
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON);
+		MockHttpServletResponse resp = mvc.perform(request).andReturn().getResponse();
+		assertTrue(resp.getStatus()==200);
+		assertTrue(resp.getContentAsString().equalsIgnoreCase(expectPostEPCostReport()));
+	}
+
+	@Test
+	public void getProdCostReportByInvalidProdName_ThenReturnResourceNotFound() throws Exception {
+		String production = "TomAndJerry";
+		when(mockService.getCostReportByProdNameAndEpCode(production, "")).thenThrow(createInvalidProductionException());
+		RequestBuilder request = MockMvcRequestBuilders
+				.get("/api/report/production/"+production)
+				.param("epCode", "")
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON);
+		MockHttpServletResponse resp = mvc.perform(request).andReturn().getResponse();
+		assertTrue(resp.getStatus()==404);
+	}
+
+	@Test
+	void postProdCostReport() throws Exception {
+		when(mockService.processProdCostReport(any(), anyList())).thenReturn(getBBTotalCostEntities());
+		RequestBuilder request = MockMvcRequestBuilders
+				.post("/api/report/production/BurntBiscuits")
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON)
+				.content(sampleBBCostInput());
+		MockHttpServletResponse resp = mvc.perform(request).andReturn().getResponse();
+		assertTrue(resp.getStatus()==200);
+		System.out.println(resp.getContentAsString());
+		System.out.println(expectPostProdCostReport());
+		assertTrue(resp.getContentAsString().equalsIgnoreCase(expectPostProdCostReport()));
+	}
+
+	@Test
+	void postAmortizedProdCostReport() throws Exception{
+		when(mockService.processAmortizedProdCostReport(any(), anyList())).thenReturn(getBBAmortizedTotalCostEntities());
+		RequestBuilder request = MockMvcRequestBuilders
+				.post("/api/report/production/BurntBiscuitsII/amortized")
+				.contentType(MediaType.APPLICATION_JSON)
+				.accept(MediaType.APPLICATION_JSON)
+				.content(sampleBBAmortizedCostInput());
+		MockHttpServletResponse resp = mvc.perform(request).andReturn().getResponse();
+		assertTrue(resp.getStatus()==200);
+		System.out.println(resp.getContentAsString());
+		System.out.println(expectAmortizedPostProdCostReport());
+		assertTrue(resp.getContentAsString().equalsIgnoreCase(expectAmortizedPostProdCostReport()));
+	}
+	
+	private String sampleMMEPCostInput() {
+		return "[{\"amount\":200.00,\"episodeCode\":\"101\"},{\"amount\":300.00,\"episodeCode\":\"101\"},{\"amount\":150.00,\"episodeCode\":\"101\"},{\"amount\":75.00,\"episodeCode\":\"101\"},{\"amount\":1325.00,\"episodeCode\":\"101\"}]";
```

Comment:
what if we need to test this with 1000 element inputs

Reply:
ok, that is a lot of records in unit test. I would probably ask you first why you need that much if for function testing? But if you really want it, I probably will pull the input out as a request.json file, and from the JUnit, I'll just read the input file in and then do rest of processing.